flink-end-to-end-tests/flink-stream-stateful-job-upgrade-test/src/main/java/org/apache/flink/streaming/tests/StatefulStreamJobUpgradeTestProgram.java

```diff
@@ -90,11 +92,13 @@ public static void main(String[] args) throws Exception {
 
     private static void executeOriginalVariant(StreamExecutionEnvironment env, ParameterTool pt)
             throws Exception {
+        Duration maxOutOfOrderness = extractTimestamp(pt);
         KeyedStream<Event, Integer> source =
                 env.addSource(createEventSource(pt))
                         .name("EventSource")
                         .uid("EventSource")
-                        .assignTimestampsAndWatermarks(createTimestampExtractor(pt))
+                        .assignTimestampsAndWatermarks(
+                                WatermarkStrategy.forBoundedOutOfOrderness(maxOutOfOrderness))
```

Comment:
After this method invocation you still need the watermark assigner here I think, same below, using the `createTimestampExtractor`

Reply:
I mean the timestamp extractor from the record

Reply:
I tried various things including 

`WatermarkStrategy.forBoundedOutOfOrderness(maxOutOfOrderness).createTimestampAssigner(createTimestampExtractor(pt)))`

and 

`WatermarkStrategy.forBoundedOutOfOrderness(maxOutOfOrderness).withTimestampAssigner(createTimestampExtractor(pt)))`

but the types never seem to match.

Not sure how to reconcile the fact that it is looking for a `TimestampAssignerSupplier` but getting a `BoundedOutOfOrdernessTimestampExtractor`.

Reply:
I think you need to changed the return type of `createTimestampExtractor`